contrib/langchain4j/src/main/java/com/google/adk/models/langchain4j/LangChain4j.java

```diff
@@ -0,0 +1,432 @@
+import com.google.genai.types.FunctionDeclaration;
+import com.google.genai.types.FunctionResponse;
+import com.google.genai.types.GenerateContentConfig;
+import com.google.genai.types.Part;
+import com.google.genai.types.Schema;
+import com.google.genai.types.ToolConfig;
+import com.google.genai.types.Type;
+import dev.langchain4j.Experimental;
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.SystemMessage;
+import dev.langchain4j.data.message.ToolExecutionResultMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.exception.UnsupportedFeatureException;
+import dev.langchain4j.model.chat.ChatModel;
+import dev.langchain4j.model.chat.StreamingChatModel;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.request.ToolChoice;
+import dev.langchain4j.model.chat.request.json.JsonArraySchema;
+import dev.langchain4j.model.chat.request.json.JsonBooleanSchema;
+import dev.langchain4j.model.chat.request.json.JsonIntegerSchema;
+import dev.langchain4j.model.chat.request.json.JsonNumberSchema;
+import dev.langchain4j.model.chat.request.json.JsonObjectSchema;
+import dev.langchain4j.model.chat.request.json.JsonSchemaElement;
+import dev.langchain4j.model.chat.request.json.JsonStringSchema;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.chat.response.StreamingChatResponseHandler;
+import io.reactivex.rxjava3.core.BackpressureStrategy;
+import io.reactivex.rxjava3.core.Flowable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+@Experimental
+public class LangChain4j extends BaseLlm {
+
+    private static final TypeReference<Map<String, Object>> MAP_TYPE_REFERENCE = new TypeReference<>() {
+    };
+
+    private final ChatModel chatModel;
+    private final StreamingChatModel streamingChatModel;
+    private final ObjectMapper objectMapper;
+
+    public LangChain4j(ChatModel chatModel) { // TODO
+        super(Objects.requireNonNull(chatModel.defaultRequestParameters().modelName(),
+            "chat model name cannot be null"));
+        this.chatModel = Objects.requireNonNull(chatModel, "chatModel cannot be null");
+        this.streamingChatModel = null;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    public LangChain4j(ChatModel chatModel, String modelName) { // TODO
+        super(Objects.requireNonNull(modelName,
+            "chat model name cannot be null"));
+        this.chatModel = Objects.requireNonNull(chatModel, "chatModel cannot be null");
+        this.streamingChatModel = null;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    public LangChain4j(StreamingChatModel streamingChatModel) { // TODO
```

Comment:
@glaforge given #175, would you potentially be open to consider "removing" the `StreamingChatModel` constructors from this PR? I've put "removing" into quotes because perhaps instead of "phyiscally removing" (deleting) them, here and also in the other 2 of the total 3 streaming related constructors, you might want to ... I don't know, just comment it out, for now? (I personally would not mind; but others may have other views about this, of course.) Or perhaps mark them e.g. `private` or `protected`? (Protected is actually a fun idea, because it would allow users to experiment with extending it...)

Reply:
I'd have to clean up the todos here and there, but see #175 for my remarks on the support of streaming. It does work for me(TM).